### ISSUE TITLE
vi binding 'e' has wrong behaviour on second-to-last char

### ISSUE THREAD

**Issue by Ambrevar** [fish-shell/fish-shell#4025]
- fish, version 2.5.0-384-g9e64571d
- OS: Arch Linux
- Term: Urxvt 9.22

Using vi bindings, if the cursor is on `1` in
```
12 345
```
pressing `e` in normal mode will move to `5` while it should move to `2`.

The current definition of `e` is

```
bind e forward-char forward-word backward-char
```

I believe the following binding to be correct:

```
bind e forward-word backward-char
```

**Comment by faho** [comment]
This was added to fix #2171.

>I believe the following binding to be correct:

>    bind e forward-word backward-char

That binding has the issue that it will get stuck on "bigwords". E.g. executing it multiple times on "12-345" will have the cursor go to "2", then "-" and then get stuck there.

We probably need to introduce additional bind functions here - "end-of-word" and "beginning-of-word".

**Comment by faho** [comment]
Also fixed by #7354.
